### DISCUSSION TITLE
how to deal with the constant growth of the logs db?

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#3861] [Q&A]
I thought this project was incredible.
pocketbase solved several problems I had while developing my micro SaaS.

I just wanted to ask a question: how to deal with growing logs? I'm having to delete the logs.db file weekly to prevent the pocketbase instance from consuming more and more RAM.

In my tests, if I let the logs accumulate, the instance starts to double its RAM consumption.

Any tips for controlling logs? Is there any way to define a validity period for the logs and delete the oldest ones?

Thank you for developing such an incredible and simple to use project.

**Comment by ganigeorgiev** [comment]
You can control the logs retention period from the _Admin UI > Logs > Logs settings > "Max days retention" input_ (setting it to `0` will disable the logs entirely).

But in general I don't think the `logs.db` size increase is directly related to your RAM consumption.

How did you measured the memory usage in the first place?

Note that the Go garbage collector will not necessary "free" its all allocated memory to the OS if the OS has plenty of ram and it doesn't need it (aka. it is not "starved").
This means that the memory usage reported by top, htop or similar tools may not exactly reflect the actual memory that the process use and it could be just "virtual"/inactive memory pages.
Practically the "scavenger" (the GC service that is responsible for releasing the memory) will occasionally notify the operating system that there are inactive memory pages that can be freed, but it is up to the OS to decide whether to reclaim them or not.

The high memory consumption could be also related to https://github.com/pocketbase/pocketbase/issues/3759 but it requires further investigation to determine whether the thumbs generation is the culprit or the S3 file uploads - https://github.com/aws/aws-sdk-go/issues/4916.


---

_Side-note: The logs are refactored for the upcoming v0.20.0 release to allow any type of logs, not just requests (available in the `develop` branch) and several optimizations have been applied, including debounced and batched writes._

_edit: Updated with the latest UI path to reflect v0.23+ changes._

**Reply by ganigeorgiev** [reply]
Alternatively, to force the GC to run more often you can also set the env variable `GOMEMLIMIT=500MiB` before starting the PocketBase process.

**Reply by erickythierry** [reply]
Thanks for the quick response.
I didn't even realize I had this logs configuration on the panel 😅 (my mistake)

Regarding RAM consumption, I didn't do a very in-depth analysis because I wasn't using it so intensely that I was worried about running out of RAM (the project is small and has low usage for now)

I use pocketbase in a docker container, and I use portainer to manage the containers and see some resource usage metrics. (I imagine these metrics are not that accurate)

but I noticed that when I restart pocketbase, it starts using less than 100mb in the first few hours. but it soon goes up to 200mb or more if I don't delete the log files from its directory. But if I delete the logs and restart, it stays below 100mb RAM usage for a few days until the logs start accumulating again.

My concern is because my project has low usage (my SaaS currently has less than 100 weekly active users) and resource usage is scaling in a way that apparently could become a problem if my project grows.

my project is a SaaS for creating bots for WhatsApp, where I host the bots on my server. I use pocketbase for login and to store user data and some bot configuration parameters (no intense read or write requests to the DB, no realtime)

When analyzing the .db files in the pb_data folder, I always notice that the log database is much larger than the database of the data I save.
So I ended up thinking that the logs were to blame for using too much RAM.

**Reply by ganigeorgiev** [reply]
@andreas-bergstrom It seems that you've deleted your comment but it will be more helpful instead of just saying _"I have the same problem"_ to provide clear details how to reproduce it because it is not even clear what actually IS the problem you are talking about in the first place and it could vary from one setup to another (especially if you are using Docker).

In any case, this discussion is very old and lot of things has changed since then but if you are still experiencing issues, please open a new Q&A discussion with more details about your environment and steps to reproduce the problem and I'll try to investigate it.

**Reply by andreas-bergstrom** [reply]
@ganigeorgiev I deleted it to save everyones time as I was asking for the log settings location in UI but found it. :) My root problem was probably that a bot caused the requests in _logs to fill up the sqlite to 3GB and since I had set max memory on the instance to 8GB it also used more RAM.
